skxray/fitting/xrf_model.py

```diff
@@ -620,70 +480,75 @@ def set_compton_model(self):
+        elastic.set_param_hint('e_offset',
+                               value=self.compton_param['e_offset'].value,
                                expr='e_offset')
-        elastic.set_param_hint('e_linear', value=self.compton_param['e_linear'].value,
+        elastic.set_param_hint('e_linear',
+                               value=self.compton_param['e_linear'].value,
                                expr='e_linear')
-        elastic.set_param_hint('e_quadratic', value=self.compton_param['e_quadratic'].value,
+        elastic.set_param_hint('e_quadratic',
+                               value=self.compton_param['e_quadratic'].value,
                                expr='e_quadratic')
-        elastic.set_param_hint('fwhm_offset', value=self.compton_param['fwhm_offset'].value,
+        elastic.set_param_hint('fwhm_offset',
+                               value=self.compton_param['fwhm_offset'].value,
                                expr='fwhm_offset')
-        elastic.set_param_hint('fwhm_fanoprime', value=self.compton_param['fwhm_fanoprime'].value,
+        elastic.set_param_hint('fwhm_fanoprime',
+                               value=self.compton_param['fwhm_fanoprime'].value,
                                expr='fwhm_fanoprime')
-        elastic.set_param_hint('coherent_sct_energy', value=self.compton_param['coherent_sct_energy'].value,
+        elastic.set_param_hint('coherent_sct_energy',
+                               value=self.compton_param['coherent_sct_energy'].value,
                                expr='coherent_sct_energy')
         logger.debug(' Finished setting up parameters for elastic model.')
         self.elastic = elastic
 
-    def set_element_model(self, ename, default_area=1e5, log_option=False):
+    def setup_element_model(self, element, default_area=1e5):
         """
         Construct element model.
 
         Parameters
         ----------
-        ename : str
-            element model
-        default_area : float
+        element : str
+            element name
+        default_area : float, optional
             value for the initial area of a given element
-        log_option : bool
-            show log or not for element setup, mainly for _set_parameter_hint function
+            default is 1e5, found to be a good value
         """
 
         incident_energy = self.incident_energy
-        parameter = self.parameter
+        parameter = self.params
 
         element_mod = None
 
-        if ename in k_line:
-            e = Element(ename)
+        if element + '_K' in K_LINE:
+            e = Element(element)
             if e.cs(incident_energy)['ka1'] == 0:
-                logger.debug(' {0} Ka emission line is not activated '
-                             'at this energy {1}'.format(ename, incident_energy))
+                logger.debug('%s Ka emission line is not activated '
+                             'at this energy %f', element, incident_energy)
```

Comment:
Because this way we lot the logging framework do the formattting.  If the
log message will never be output (for example if this logger's level is set
above debug) the formatting operation never takes place which saves us a
bit of work.

On Thu, Mar 26, 2015 at 3:16 PM Li Li <notifications@github.com> wrote:

> In skxray/fitting/xrf_model.py
> <https://github.com/licode/scikit-xray/pull/6#discussion_r27246373>:
>
> >              if e.cs(incident_energy)['ka1'] == 0:
> > -                logger.debug(' {0} Ka emission line is not activated '
> > -                             'at this energy {1}'.format(ename, incident_energy))
> > +                logger.debug('%s Ka emission line is not activated '
> > +                             'at this energy %f', element, incident_energy)
>
> why not use format?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/licode/scikit-xray/pull/6/files#r27246373>.
>